addons/pos_pinelabs/static/src/app/payment_pinelabs.js

```diff
@@ -0,0 +1,242 @@
+import { _t } from "@web/core/l10n/translation";
+import { PaymentInterface } from "@point_of_sale/app/payment/payment_interface";
+import { AlertDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { serializeDateTime } from "@web/core/l10n/dates";
+
+const REQUEST_TIMEOUT = 10000;
+const CANCEL_REQUEST_TIME_LIMIT = 125000;
+const { DateTime } = luxon;
+
+export class PaymentPinelabs extends PaymentInterface {
+    setup() {
+        super.setup(...arguments);
+        this.paymentInitialTime = null;
+        this.pollingTimeout = null;
+        this.inactivityTimeout = null;
+        this.queued = false;
+        this.payment_stopped = false;
+    }
+
+    send_payment_request(cid) {
+        super.send_payment_request(cid);
+        return this._processPinelabs();
+    }
+
+    pendingPinelabsPaymentLine() {
+        return this.pos.getPendingPaymentLine("pinelabs");
+    }
+
+    send_payment_cancel(order, cid) {
+        super.send_payment_cancel(order, cid);
+        return this._pinelabsCancel();
+    }
+
+    _callPinelabs(data, action) {
+        return this.env.services.orm.silent
+            .call("pos.payment.method", action, [[this.payment_method_id.id], data])
+            .catch(this._handle_odoo_connection_failure.bind(this));
+    }
+
+    _handle_odoo_connection_failure(data = {}) {
+        // handle timeout
+        const line = this.pendingPinelabsPaymentLine();
+        if (line) {
+            line.set_payment_status("retry");
+        }
+        this._showError(
+            _t(
+                "Unable to connect to the Odoo server. Please check your internet connection and try again."
+            )
+        );
+
+        return Promise.reject(data); // prevent subsequent unFulFilled's from being called
```

Comment:
just because there are some stupid patterns in `pos_adyen` it doesn't mean that we have to propagate them in every single payment module that we now create ( for ex razorpay and viva wallet and apparently now pinelabs ).

We are even completely refactoring `pos_adyen` at the moment, precisely because it's weirdly written. 

If you don't have a good reason for having a line in your pr please don't just copy paste it there.

There is no problem with promises and they are perfectly fine to be used, but we have to have consistency and logic. If you mix new code with old code from other modules, promises and async/await and all sorts of other things the result will be a mess.

Please take a moment to rethink this file completely.

Some pointers:
1. Don't use `pos_adyen` as inspiration. It's _not_ an example of good code.
2. Think deeply whether or not you need all of the things that you have here .
3. Refactor to async/await ( not because promises are bad, but because it's weird to be constantly switching between mechanisms; everything becomes harder to understand )